core/controllers/resources.py

```diff
@@ -76,6 +77,11 @@ def get(self, entity_type, entity_id, asset_type, encoded_filename):
             self.response.headers['Content-Type'] = str(
                 '%s/%s' % (asset_type, file_format))
 
+            if entity_type == feconf.ENTITY_TYPE_SUBTOPIC:
```

Comment:
This seems odd. The entity_type passed in should be topic; subtopic should not be supported here.

Reply:
Maybe you could change the signature to 

     get(self, page_context, page_identifier, ...)

and then translate (page_context, page_identifier) to (entity_type, entity_id)

Reply:
Done